packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -361,62 +361,102 @@ it('should handle checkbox clicks', async () => {
 it('should handle row clicks', async () => {
   const onSelect = jest.fn();
   const onRowClick = jest.fn();
+  const user = userEvent.setup();
+  const data = mockedData(5);
+
   const { container, getByText } = renderComponent({
     isSelectable: true,
     onSelect,
     onRowClick,
+    data,
   });
 
   expect(screen.queryByText('Header name')).toBeFalsy();
   const rows = container.querySelectorAll('.iui-table-body .iui-row');
-  expect(rows.length).toBe(3);
-
-  const user = userEvent.setup();
+  expect(rows.length).toBe(5);
 
-  // Shift click test #1
+  // Shift click special case test #1
   // By default, when no row is selected before shift click, start selecting from first row to clicked row
-  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
-  await user.click(getByText(mockedData()[1].name)); // [shiftKey: true]
+  await user.keyboard('{Shift>}'); // Hold Shift
+  await user.click(getByText(data[2].name));
 
   expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).toContain('iui-selected');
-  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(rows[3].classList).not.toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(1);
   expect(onRowClick).toHaveBeenCalledTimes(1);
 
-  await userEvent.click(getByText(mockedData()[1].name)); // Deselect
+  await user.keyboard('{/Shift}'); // Release Shift
+  await userEvent.click(getByText(data[1].name)); // Deselect; lastSelectedRowId = undefined -> 1
+
   expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).not.toContain('iui-selected');
-  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(rows[3].classList).not.toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(2);
   expect(onRowClick).toHaveBeenCalledTimes(2);
 
-  await userEvent.click(getByText(mockedData()[1].name)); // Reselect; lastSelectedRowId = undefined -> 1
-  expect(rows[0].classList).not.toContain('iui-selected');
+  // Shift click special case test #2
+  // When a row is clicked before shift click (lastSelectedRowId), selection starts from that row and ends at the currently clicked row
+  // But if the startIndex > endIndex, then startIndex and endIndex are swapped
+  // (Here startIndex = 1, endIndex = 0)
+  await user.keyboard('{Shift>}'); // Hold Shift
+  await user.click(getByText(data[0].name));
+
+  expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).toContain('iui-selected');
   expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[3].classList).not.toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(3);
   expect(onRowClick).toHaveBeenCalledTimes(3);
```

Comment:
I'm performing this test on the table story, but it's not doing the same thing as in the tests. So from the beginning of this test I shiftclick #2 -> click #1 -> shiftclick #0 -> controlclick #3 -> shiftclick #4 -> click checkbox#1 -> shiftclick #3. The third action, shiftclick #0 (after deselecting #1), does not select rows #0 & #1, just #0.


https://user-images.githubusercontent.com/37936169/199308008-c9e5bd68-0c3d-49c5-b43b-64a83bf8774c.mp4

Reply:
Not sure why that's happening. I was able to make it work correctly on my end.

[Video4.webm](https://user-images.githubusercontent.com/45748283/199328257-6c6f6970-e067-4f88-a215-53523238475e.webm)

To confirm, are you testing the `rohan/table-shift-click-checkbox-bug-1` branch? Because there is another branch with a similar name (`rohan/table-ctrl-shift-click-1`)

Reply:
Hm, maybe I need to update it

Reply:
That was it. Everything seems fine now!